### DISCUSSION TITLE
Proper way to do cron backup?

### DISCUSSION THREAD

**Discussion by RonsonDk** [teslamate-org/teslamate#3185] [Q&A]
I recently reinstalled my teslamate VM and restored the data, using the documentation (docker on Ubuntu 22).

On the new VM I copied my old backup script but I can't get it to work. On the old setup I ran everything as root, on the new one I use my '_ronson_' user. The script looks like this:

```
#!/bin/bash

LOG=/home/ronson/backup_log
NOW=$(date '+%F')

cd /home/ronson

#Debug:
echo "User:" $(whoami) >> ${LOG}
echo "Dir:" $(pwd) >> ${LOG}

echo "Starting:" $(date) >> ${LOG}
docker-compose exec -T database pg_dump -U teslamate teslamate > ${NOW}_teslamate.bck

echo "Delete old backups:" $(date) >> ${LOG}
rm 202*.bck.gz

echo "Gzip:" $(date) >> ${LOG}
gzip --best ~/${NOW}_teslamate.bck >> ${LOG} 2>&1

echo "Done:" $(date) >> ${LOG}
```

When I run it manually (as '_ronson_') it works just fine.
But if I run it via cron (also as '_ronson_' according to the log) the DB backup always only contains:

> Error: No such container: 03b7d598d99a0e27832613cfd5f9c1aabca90a38e2ef431e20d63bd842c6cba1

What am I doing wrong?
Sooner or later I will forget to backup manually..

**Comment by DrMichael** [comment]
Is your `docker-compose.yml` really in /home/ronson?
I use the following `backup.sh`, `docker-compose.yml` is in `${HOME}/teslamate`:
```
#!/bin/sh

WEEKDAY=`/bin/date +%A`

cd ${HOME}/teslamate
docker-compose exec -T database pg_dump -U teslamate teslamate | gzip > backups/teslamate-${WEEKDAY}.bck.gz
```

**Reply by RonsonDk** [reply]
Yes, it is in /home/ronson:
```
ronson@teslamate:~$ pwd
/home/ronson
ronson@teslamate:~$ ls doc*
docker-compose.yml
```
backup_log shows the same directory and user when started via cron:

> User: ronson
> Dir: /home/ronson

**Comment by CarlosCuezva** [comment]
Can you show the crontab line?

**Reply by RonsonDk** [reply]
I ran (as '_ronson_')
`crontab -e`
and here I added
`30 0 * * * /home/ronson/backup`

The script runs at the correct time.

**Reply by RonsonDk** [reply]
'_ronson_' is a member of the group:
```
ronson@teslamate:~$ getent group docker
docker:x:1001:ronson
```

**Reply by CarlosCuezva** [reply]
Test it with this change

```
30 0 * * * sh /home/ronson/backup
```
